build/Build.Steps.Windows.cs

```diff
@@ -118,6 +120,50 @@ partial class Build
             );
         });
 
+    Target GenerateNetFxTransientDependencies => _ => _
+        .Unlisted()
```

Comment:
This target is unlisted and not used anywhere. Shouldn't it be invoked in GH Actions? Otherwise we will have missing entries. Personally I think we could add it to the `Workflow`. We have a `git diff` in `ci.yml - build` GH job that would make sure that everything is up to date.

Reply:
I agree with this: unless it is relatively slow I would say that it should be executed before the `CompileManagedSrc` when the TFM is `net462`. If it is really slow it could be part of the `Workflow`.

Reply:
I currently don't see any need to auto execute it. It should be executed only within a PR that is going to break dependencies.
Secondly it will generate uncommitted changes then, but all of these changes must be committed for dependendabot to work.

Reply:
> It should be executed only within a PR that is going to break dependencies.

How we will know it if it is not automated?

> Secondly it will generate uncommitted changes then, but all of these changes must be committed for dependendabot to work.

Then we will just push commits to the dependabot PRs. Thanks to it we will make sure that `main` is always "clean".

Reply:
The goal is to have the list of assemblies automatically update whenever needed. It seems to me that the simplest way to achieve that seems to have this step as part of `BuildTracer` target.

Reply:
> How we will know it if it is not automated?
That's exactly why the test is there. It enforces you to run `GenerateNetFxTransientDependencies` just in the right time, so external dependencies are clearly visible and connected to the PR that caused these. 
Running it fully automated causes dependencies to be added without any clear source trail.

Reply:
A test covering this is good too but as of right now if the csproj gets updated due to some other change CI will flag it so no accidental changes are going to go through one way or another. If you have it automated as part of the build the change will be visible locally even before the file is submitted so the author can review it.

Reply:
That actually makes sense, I originally thought that the target is somewhat expensive to be in the main pipeline. 
On my desktop it's just 2 seconds (looking at nuke's feedback), will see after a CI run as it's the weakest machine.

Reply:
well `GenerateNetFxTransientDependencies Succeeded  0:13` not too terrible actually.

Reply:
Excellent! Much better than what I expected too.